main.tf

```diff
@@ -242,6 +247,8 @@ resource "azurerm_network_security_rule" "vm-ssh" {
   destination_address_prefix  = "*"
   resource_group_name         = azurerm_resource_group.azure_rg.name
   network_security_group_name = azurerm_network_security_group.nsg.name
+
+  depends_on = [module.nfs]
```

Comment:
What is the reason for adding this dependency on the nfs module? This rule is for all vm's created. Not sure why this is needed.

Reply:
To add explicit dependencies. There were collision with priority values, likely from prior to merge that merged nfs & jump nsg rules. I can test and remove.